app/src/main/java/org/elasticsearch/bootstrap/BootstrapProxy.java

```diff
@@ -107,50 +108,17 @@ public class BootstrapProxy {
     /**
      * initialize native resources
      */
-    static void initializeNatives(boolean mlockAll, boolean ctrlHandler) {
-        final Logger logger = LogManager.getLogger(BootstrapProxy.class);
-
+    static void initializeNatives(boolean mlockAll) {
         // check if the user is running as root, and bail
         if (Natives.definitelyRunningAsRoot()) {
             throw new RuntimeException("can not run crate as root");
         }
 
         // mlockall if requested
-        if (mlockAll) {
-            if (Constants.WINDOWS) {
-                Natives.tryVirtualLock();
-            } else {
-                Natives.tryMlockall();
-            }
-        }
-
-        // listener for windows close event
-        if (ctrlHandler) {
-            Natives.addConsoleCtrlHandler(code -> {
-                if (ConsoleCtrlHandler.CTRL_CLOSE_EVENT == code) {
-                    logger.info("running graceful exit on windows");
-                    try {
-                        BootstrapProxy.stop();
-                    } catch (IOException e) {
-                        throw new ElasticsearchException("failed to stop node", e);
-                    }
-                    return true;
-                }
-                return false;
-            });
-        }
-
-        // force remainder of JNA to be loaded (if available).
-        try {
-            JNAKernel32Library.getInstance();
-        } catch (Exception ignored) {
-            // we've already logged this.
+        if (mlockAll && !Constants.WINDOWS) {
+            Natives.tryMlockall();
         }
 
-        Natives.trySetMaxNumberOfThreads();
```

Comment:
These weren't actively changing system settings but retrieving the values and setting a static variable as side effect. I changed this so the bootstrap checks query the current values directly.

Reply:
> These weren't actively changing system settings but retrieving the values and setting a static variable as side effect. I changed this so the bootstrap checks query the current values directly.

I've tested this on MacOS 15.4.1 (Darwin Kernel Version 24.4.0) and all worked fined.